### PR TITLE
Equality

### PR DESCRIPTION
**Pull request by marianogonzalez**

### CODE REVIEW

File: src/main/java/org/mule/extension/oauth2/api/authorizationcode/DefaultAuthorizationCodeGrantType.java [hunk 1]
```diff
@@ -151,6 +156,33 @@
   @Optional(defaultValue = "true")
   private boolean encodeClientCredentialsInBody;
 
+  @Override
+  public boolean equals(Object obj) {
+    if (obj instanceof DefaultAuthorizationCodeGrantType) {
```
Comment (afelisatti) [inline src/main/java/org/mule/extension/oauth2/api/authorizationcode/DefaultAuthorizationCodeGrantType.java]: Doesn't this have a super class with more stuff to consider?

File: src/main/java/org/mule/extension/oauth2/internal/AbstractGrantType.java [hunk 2]
```diff
@@ -206,6 +209,36 @@ public void setMuleContext(MuleContext muleContext) {
...
+          Objects.equals(tlsContextFactory, other.tlsContextFactory) &&
+          Objects.equals(proxyConfig, other.proxyConfig);
+    }
+
+    return false;
+  }
+
+  @Override
+  public int hashCode() {
+    return 31 * hash(
```
Comment (afelisatti) [inline src/main/java/org/mule/extension/oauth2/internal/AbstractGrantType.java]: Does the formatter like this? OMG

File: src/main/java/org/mule/extension/oauth2/internal/OAuthUtils.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+import org.mule.runtime.extension.api.runtime.parameter.ParameterResolver;
+
+import java.util.Objects;
+
+public final class OAuthUtils {
+
+  private OAuthUtils() {}
+
+  public static <T> boolean literalEquals(Literal<T> l1, Literal<T> l2) {
+    if ((l1 == null && l2 == null)) {
```
Comment (afelisatti) [inline src/main/java/org/mule/extension/oauth2/internal/OAuthUtils.java]: extra ()

File: src/main/java/org/mule/extension/oauth2/internal/OAuthUtils.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+    return Objects.equals(l1.getLiteralValue().orElse(null), l2.getLiteralValue().orElse(null));
+  }
+
+  public static <T> int literalHashCode(Literal<T> literal) {
+    return literal != null ? literal.getLiteralValue().map(Object::hashCode).orElse(0) : 0;
+  }
+
+  public static int literalHashCodes(Literal... literals) {
+    if (literals == null)
```
Comment (afelisatti) [inline src/main/java/org/mule/extension/oauth2/internal/OAuthUtils.java]: The formatter liking this is a bug, use braces. In the for as well.

File: src/main/java/org/mule/extension/oauth2/internal/OAuthUtils.java [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+    for (Literal literal : literals)
+      result = 31 * result + (literal == null ? 0 : literalHashCode(literal));
+
+    return result;
+
+  }
+
+  public static <T> boolean resolverEquals(ParameterResolver<T> p1, ParameterResolver<T> p2) {
+    if ((p1 == null && p2 == null)) {
```
Comment (afelisatti) [inline src/main/java/org/mule/extension/oauth2/internal/OAuthUtils.java]: extra ()

File: src/main/java/org/mule/extension/oauth2/api/authorizationcode/DefaultAuthorizationCodeGrantType.java (hunk 1 above)
Comment (marianogonzalez) [reply src/main/java/org/mule/extension/oauth2/api/authorizationcode/DefaultAuthorizationCodeGrantType.java]: good catch

File: src/main/java/org/mule/extension/oauth2/internal/AbstractGrantType.java (hunk 2 above)
Comment (marianogonzalez) [reply src/main/java/org/mule/extension/oauth2/internal/AbstractGrantType.java]: it not only likes it, it changed what I did manually in favor of this beauty

Comment (afelisatti) [reply src/main/java/org/mule/extension/oauth2/internal/AbstractGrantType.java]: :wheaton:

**Pull request merged**